test/libyul/yulOptimizerTests/fullSuite/unusedFunctionParameterPruner_loop.yul

```diff
@@ -0,0 +1,33 @@
+{
+    sstore(f(1), 1)
+    sstore(f(2), 1)
+    sstore(f(3), 1)
+    function f(a) -> x {
+        for {let b := 10} iszero(b) { b := sub(b, 1) }
+        {
+            a := calldataload(0)
+            mstore(a, x)
+        }
+    }
+}
+// ----
+// step: fullSuite
+//
+// {
+//     {
+//         f()
+//         sstore(0, 1)
```

Comment:
Does it  infer the result of `f` to be always zero from the 'unused return value pruning'? But this pruning is not part of the pull request yet, is it?

Reply:
We haven't used 'callsite' optimization yet. The value is `0` because of the inlining.

Reply:
Oh, because `x` is only read from and not written to?